transitions/registration.js

```diff
@@ -217,21 +216,29 @@ module.exports = {
                 if (self.isBoolExprFalse(obj, event.bool_expr)) {
                     return;
                 }
-                var options = { db: db, audit: audit, doc: doc };
-                if (event.params) {
-                    // params setting get sent as array
+                var options = { db: db, audit: audit, doc: doc, registrationConfig: registrationConfig };
+
+                if (typeof event.params === 'string') {
+                    // params setting can get sent as string to convert into an array
                     options.params = event.params.split(',');
+                } else if (typeof event.params === 'object') {
```

Comment:
I don't think there's a JSON type. There's an object type but that expects the object to be defined not freetext. I've been thinking for a while we should deprecate the comma separated params in favour of JSON array (or object in this case) style, so we could add a second property for json_params (come up with a better name)? Or just take the string and JSON.parse, catch the error, try splitting, give up?